### PR TITLE
Added tests for sunpy/net/hek2vso/hek2vso.py

### LINKED ISSUE #2970: Increase test coverage of sunpy.net.{hek,hek2vso,helio}
These three net submodules (hek, hek2vso, helio) need more tests written for them.

Coverage at time of writing:

* HEK: [14.85%](https://codecov.io/gh/sunpy/sunpy/tree/master/sunpy/net/hek)
* HEK2VSO: [55.38%](https://codecov.io/gh/sunpy/sunpy/tree/master/sunpy/net/hek2vso)
* HELIO: [46.15%](https://codecov.io/gh/sunpy/sunpy/tree/master/sunpy/net/helio), this is for the client file.

Also these tests should be moved to the correct folders as well.

### PR DESCRIPTION
**Pull request by jains8844**
Added tests for sunpy/net/hek2vso/hek2vso.py

With reference to  #2970

### CODE REVIEW

**Comment by jains8844** [comment]
Why are all these checks failing ?

**Comment by nabobalis** [comment]
It would seem your changes broke something. Did you run the tests locally?

**Comment by nabobalis** [comment]
Seems like something is wrong with this fixture:

```
@pytest.mark.remote_data
    @pytest.fixture(scope="function")
    def h2v_client():
>       return hek2vso.H2VClient()
```

**Comment by jains8844** [comment]
The tests are getting passed locally for this file.

**Comment by nabobalis** [comment]
The full traceback is:

```
[gw0] linux -- Python 3.6.8 /home/vsts/work/1/s/.tox/conda/bin/python

    @pytest.mark.remote_data
    @pytest.fixture(scope="function")
    def h2v_client():
>       return hek2vso.H2VClient()

../.tox/conda/lib/python3.6/site-packages/sunpy/net/tests/test_hek2vso.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../.tox/conda/lib/python3.6/site-packages/sunpy/net/hek2vso/hek2vso.py:126: in __init__
    self.vso_client = vso.VSOClient()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sunpy.net.vso.vso.VSOClient object at 0x7f1d510fcf60>, url = None, port = None, api = None

    def __init__(self, url=None, port=None, api=None):
        api = get_online_vso_url(api, url, port)
        if api is None:
>           raise ConnectionError("Cannot find an online VSO mirror.")
E           ConnectionError: Cannot find an online VSO mirror.
```

If it works locally, I wonder why it fails on the servers. Can you run the test suite in tox locally and see if that errors too?

**Comment by jains8844** [comment]
Okay I'm running. Will update once I'm done with that.

**Comment by jains8844** [comment]
The tests are failing for offline one. Can't I have more than one pytest decorator for a function ?

**Comment by nabobalis** [comment]
You should be able to

**Comment by jains8844** [comment]
Yeah got it. I missed a marker for a class function.

**Comment by jains8844** [comment]
I've fixed the code still the checks are failing ? 
The errors displayed in the checks doesn't belong to this test I guess

**Comment by nabobalis** [comment]
Yes the errors seem to be unrelated to the PR. The servers should be back now.

File: sunpy/net/tests/test_hek2vso.py [hunk 1]
```diff
@@ -61,28 +64,98 @@ def test_translate_results_to_query():
...
+            )
+        )
+
+        assert h2v.num_of_records != 0
+        assert len(h2v.vso_results) != 0
+
+        h2v._quick_clean()
+
+        assert h2v.vso_results == []
+        assert h2v.num_of_records == 0
```
Comment (nabobalis) [inline sunpy/net/tests/test_hek2vso.py]: Was this formatted by yapf?

Comment (jains8844) [reply sunpy/net/tests/test_hek2vso.py]: No, I gave this line space to separate other statements from assert ones.

**Comment by jains8844** [comment]
Are there some changes required in this PR ?

File: sunpy/net/tests/test_hek2vso.py [hunk 2]
```diff
@@ -61,28 +64,98 @@ def test_translate_results_to_query():
...
+        hek_query[0]["obs_wavelunit"]
+    )
+    assert vso_query[3].max == hek_query[0]["obs_meanwavel"] * u.Unit(
+        hek_query[0]["obs_wavelunit"]
+    )
+    assert vso_query[3].unit == u.Unit("Angstrom")
 
 
+@pytest.mark.remote_data
 class TestH2VClient(object):
```
Comment (nabobalis) [inline sunpy/net/tests/test_hek2vso.py]: Can you undo this class structure?

Comment (jains8844) [reply sunpy/net/tests/test_hek2vso.py]: Means ?

Comment (nabobalis) [reply sunpy/net/tests/test_hek2vso.py]: Can you rewrite the tests to not use this class?

Comment (jains8844) [reply sunpy/net/tests/test_hek2vso.py]: Yup. I can have the functions written in the class separately right ?

Comment (nabobalis) [reply sunpy/net/tests/test_hek2vso.py]: You can remove the functions from the class sure.

Comment (jains8844) [reply sunpy/net/tests/test_hek2vso.py]: Like this ?

Comment (nabobalis) [reply sunpy/net/tests/test_hek2vso.py]: Yes

**Comment by nabobalis** [comment]
Thanks @jains8844

**Pull request merged**